tools/src/main/java/com/graphhopper/tools/Measurement.java

```diff
@@ -304,11 +304,10 @@ private GraphHopperConfig createConfigFromArgs(PMap args) {
         ghConfig.setCHProfiles(chProfiles);
         List<LMProfileConfig> lmProfiles = new ArrayList<>();
         if (useLM) {
-            // as currently we do not allow cross-querying LM with turn costs=true/false we have to add both
-            // profiles and this currently leads to two identical LM preparations
             lmProfiles.add(new LMProfileConfig("profile_no_tc"));
             if (turnCosts)
-                lmProfiles.add(new LMProfileConfig("profile_tc"));
+                // no need for a second LM preparation, we can do cross queries here
+                lmProfiles.add(new LMProfileConfig("profile_tc").setPreparationProfile("profile_no_tc"));
```

Comment:
This is a good use-case for the new feature: We use the same landmarks for both LM profiles (with and without turn costs).